test/functional/tests/initialize/test_startup_init_config.py

```diff
@@ -160,21 +163,97 @@ def test_cas_startup_lazy():
       - DUT boots sucesfully
       - caches are configured as expected
-    steps:
-      - Prepare one drive for caches and one for cores
-      - Create 2 cache partitions and 4 core partitons
-      - Create opencas.conf config for 2 caches each with 2 core partition as cores
-      - Mark first cache as lazy_startup=True
-      - Mark first core of second cache as lazy_startup=True
-      - Run casctl init
-      - Run casctl stop
-      - Remove first cache partition
-      - Remove first core of second cache partition
-      - Reboot DUT
-      - Verify DUT booted successfully
-      - Verify CAS configured properly
     """
-    pass
+    with TestRun.step("Prepare partitions"):
+        cache_disk = TestRun.disks["cache"]
+        core_disk = TestRun.disks["core"]
+        cache_disk.create_partitions([Size(200, Unit.MebiByte)] * 2)
+        core_disk.create_partitions([Size(200, Unit.MebiByte)] * 4)
+
+    with TestRun.step(f"Add a cache configuration with cache device with `lazy_startup` flag"):
+        init_conf = InitConfig()
+        init_conf.add_cache(1, cache_disk.partitions[0], extra_flags="lazy_startup=True")
+        init_conf.add_core(1, 1, core_disk.partitions[0])
+        init_conf.add_core(1, 2, core_disk.partitions[1])
+
+        expected_core_pool_paths = set(c.path for c in core_disk.partitions[:2])
+
+    with TestRun.step(f"Add a cache configuration with core device with `lazy_startup` flag"):
+        init_conf.add_cache(2, cache_disk.partitions[1])
+        init_conf.add_core(2, 1, core_disk.partitions[2])
+        init_conf.add_core(2, 2, core_disk.partitions[3], extra_flags="lazy_startup=True")
+        init_conf.save_config_file()
+        sync()
+
+        expected_caches_paths = set([cache_disk.partitions[1].path])
+        expected_cores_paths = set(c.path for c in core_disk.partitions[2:])
+        active_core_path = core_disk.partitions[2].path
+        inactive_core_path = core_disk.partitions[3].path
+
+    with TestRun.step(f"Start and stop all the configurations using the casctl utility"):
+        output = casctl.init(True)
+        if output.exit_code != 0:
+            TestRun.fail(f"Failed to initialize caches from config file. Error: {output.stdout}")
+        casadm.stop_all_caches()
+
+    with TestRun.step(
+        "Disable udev to allow manipulating partitions without CAS being automatically loaded"
+    ):
+        Udev.disable()
+
+    with TestRun.step(f"Remove one cache patition and one core partition"):
+        cache_disk.remove_partition(cache_disk.partitions[0])
+        core_disk.remove_partition(core_disk.partitions[3])
+
+    with TestRun.step("Reboot DUT"):
+        power_control = TestRun.plugin_manager.get_plugin("power_control")
+        power_control.power_cycle()
+
+    with TestRun.step("Verify if all the devices are initialized properly"):
+        core_pool_list = get_cas_devices_dict()["core_pool"]
+        caches_list = get_cas_devices_dict()["caches"].items()
```

Comment:
since you are using only values from this dictionary you can use `.values()` getter